Br pub exit issue

muhamadazmy opened the pull request:
This PR includes
- Fix public exit nic detection, errors where not handled correctly which caused some nics to be selected when they shouldn't
- Show the current node setup on UI (single, dual) and what nic is selected
- Add authorized (only the farmer can make) calls to the node rmb interface:
  - List all physical devices on the node. 
  - Choose to switch which interface is used by br-pub as an exit interface. The choices can be either `zos` to fall back to single nic setup, or any other free physical nic.
  -

muhamadazmy committed:
Fix issue for nic selection

muhamadazmy committed:
better debugging

muhamadazmy committed:
Updage zbus to latest version

muhamadazmy committed:
Fix failed tests

muhamadazmy committed:
WIP: support chaning br-pub interface

muhamadazmy committed:
WIP: add new Admins middleware for rmb

muhamadazmy committed:
regenerate stubs

muhamadazmy committed:
Build rmb api for network admin mgmt

muhamadazmy committed:
finalize the br-pub exit nic setter

muhamadazmy committed:
handle initialization error

muhamadazmy committed:
Make sure root is initialized on boot

muhamadazmy committed:
Improve the API to manage public link

LeeSmet commented on pkg/network/public/public.go:
```diff
@@ -100,15 +102,117 @@ func setupPublicBridge(br *netlink.Bridge) error {
+
+// SetPublicExitLink rewires the br-pub to a different exit (upstream) device.
+// this upstream device can either be a physical free device, or zos bridge.
+// the method does nothing if
```
Incomplete docstring

LeeSmet commented on cmds/modules/zui/net.go:
```diff
@@ -2,6 +2,7 @@ package zui
 import (
 	"context"
+	"fmt"
 	_ "fmt"
```
double import of "fmt"

LeeSmet reviewed (dismissed):
looks good, see comments

muhamadazmy committed:
Add comments

muhamadazmy committed:
avoid possible missing of events

We check the chain every 2 hours anyway
for changes in public config

muhamadazmy committed:
Make sure we persiste current uplink on update

muhamadazmy committed:
Update PR comments

muhamadazmy committed:
remove invalid filter, a nic can have slaac address

Better to remove this filter in case the nic got an ip
immediately after bringing it up.
